third_party/nvfuser/csrc/python_frontend/fusion_definition.h

```diff
@@ -72,6 +74,69 @@ struct TORCH_CUDA_CU_API Scalar {
   size_t index;
 };
 
+//! Fusion State contains the information used to build a new cpp Fusion object.
+class FusionState {
+ public:
```

Comment:
I would put the `FusionState` class in a separate file (ie `fusion_state.h` and `fusion_state.cpp`).